### PR TITLE
Object Pooling ... again

### PR DESCRIPTION
**Pull request by eswordthecat**
[] merge clusterfack pooling datum with my old ones
[x] pool light source
[] pool signal

### CODE REVIEW

File: code/controllers/Processes/lighting.dm [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 	for(var/datum/light_source/L in lighting_controller.lights)
 		if(L && L.check())
-			lighting_controller.lights.Remove(L)
```
Comment (ComicIronic) [inline code/controllers/Processes/lighting.dm]: Don't you have to keep the Remove because this doesn't remove the ref otherwise?

Comment (eswordthecat) [inline code/controllers/Processes/lighting.dm]: It's on Destroy now.

**Comment by PJB3005** [comment]
Still not sure how usefull adding anything to DAL is since it's being removed

**Comment by ComicIronic** [comment]
RIP this PR

**Pull request closed without merging**